### PR TITLE
Fix rec tests

### PR DESCRIPTION
**Pull request by ChristineStawitz-NOAA**
Thanks for opening a pull request (PR) to FIMS! Please describe the changes under the headings below. Please also go through the checklist below to ensure the appropriate steps have been taken during development. For smaller changes, feel free to skip steps that aren't relevant, but at least describe the feature under the "what is the feature".

# What is the feature?
Fixes the recruitment nll test which is now in R

# How have you implemented the solution?
Replaced the C++ test for recruitment nll with an R test since the nll module depends on TMB

# Does it impact any other area of the project?
Test cases and population

# Does this change impact the model input or output?*
This allows bias correction and negative log likelihood for rec devs to be done

# How to test this change
test-recruitment-interface.R

# Developer pre-PR Checklist

Please do these steps locally for big changes. Note GitHub Actions does all of these checks automatically when pushing to the repository. Please see [code development section in the contributor guide](https://noaa-fims.github.io/collaborative_workflow/contributor-guidelines.html#code-development) for details.

- [x] Run [cmake build and ctest locally](https://noaa-fims.github.io/collaborative_workflow/testing.html#c-unit-testing-and-benchmarking) and make sure the C++ tests pass
- [ ] Run `devtools::document()` locally and push changes to the remote feature branch
- [ ] Run `styler::style_pkg()` locally and push changes to remote feature branch. If there are many changes, please do this in a separate commit.
- [x] Run `devtools::check()` locally and make sure the package can be compiled and R tests pass. If there are failing tests, run `devtools::test(filter = "file_name")` (where "test-file_name.R" is the testthat file containing failing tests) and edit code/tests to troubleshoot tests.
- [x] Before opening the PR, make sure all the github actions are passing on the remote feature branch.
- [x] Make sure this PR has an informative title.

### CODE REVIEW

File: inst/include/interface/rcpp/rcpp_objects/rcpp_nll.hpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+};
+
+uint32_t NLLInterfaceBase::id_g = 1;
+std::map<uint32_t, NLLInterfaceBase*>
+    NLLInterfaceBase::live_objects;
+
+ class RecruitmentNLLInterface : public NLLInterfaceBase {
+ public:
+  Parameter log_sigma_recruit;   /**< log of the sd recruitment*/
+  Rcpp::NumericVector recruit_deviations; /**< vector of recruitment devs*/
```
Comment (msupernaw) [inline inst/include/interface/rcpp/rcpp_objects/rcpp_nll.hpp:46]: should be using RCpp::Vector

File: inst/include/interface/rcpp/rcpp_objects/rcpp_nll.hpp [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  Rcpp::NumericVector recruit_bias_adjustment; /**<vector bias adjustment*/
```
Comment (msupernaw) [inline inst/include/interface/rcpp/rcpp_objects/rcpp_nll.hpp:47]: should be using RCpp::Vector here

**Comment by Andrea-Havron-NOAA** [comment]
should we merge this first into 180-feat-population?

**Comment by k-doering-NOAA** [comment]
@Andrea-Havron-NOAA is right, I think the intention was to merge this into 180-feat-population rather than main (see compare: https://github.com/NOAA-FIMS/FIMS/compare/180-feat-population...fix-rec-tests)

**Review by MOshima-PIFSC — APPROVED** [review]
Changes look good to me.

**Pull request merged**